### ISSUE TITLE
curl: (52) Empty reply from server

### ISSUE THREAD

**Issue by laktak** [gliderlabs/docker-consul#100]
I always get an empty response from consul:

```
# curl localhost:8500/v1/catalog/nodes`
curl: (52) Empty reply from server

# docker exec consul consul members
Node     Address              Status  Type    Build  Protocol  DC
dev-do1  10.246.246.174:8301  alive   server  0.5.2  2         dc1
dev-do3  10.246.246.177:8301  alive   server  0.5.2  2         dc1
dev-do2  10.246.246.173:8301  alive   server  0.5.2  2         dc1

# docker exec consul consul info
WARNING: It is highly recommended to set GOMAXPROCS higher than 1

agent:
    check_monitors = 0
    check_ttls = 0
    checks = 0
    services = 1
build:
    prerelease = 
    revision = 9a9cc934
    version = 0.5.2
consul:
    bootstrap = false
    known_datacenters = 1
    leader = true
    server = true
raft:
    applied_index = 17
    commit_index = 17
    fsm_pending = 0
    last_contact = never
    last_log_index = 17
    last_log_term = 1
    last_snapshot_index = 0
    last_snapshot_term = 0
    num_peers = 2
    state = Leader
    term = 1
runtime:
    arch = amd64
    cpu_count = 2
    goroutines = 72
    max_procs = 1
    os = linux
    version = go1.4.2
serf_lan:
    encrypted = false
    event_queue = 0
    event_time = 2
    failed = 0
    intent_queue = 0
    left = 0
    member_time = 3
    members = 3
    query_queue = 0
    query_time = 1
serf_wan:
    encrypted = false
    event_queue = 0
    event_time = 1
    failed = 0
    intent_queue = 0
    left = 0
    member_time = 1
    members = 1
    query_queue = 0
    query_time = 1
```

Started with

```
docker run --restart=always -d --name consul -v /opt/consul:/opt/consul -p 8300:8300 -p 8301:8301 -p 8301:8301/udp -p 8302:8302 -p 8302:8302/udp -p 8400:8400 -p 8500:8500 -p 8600:53/udp gliderlabs/consul agent -server -data-dir=/opt/consul/data -config-dir=/opt/consul/config -advertise=10.246.246.174 -node=dev-do1 -bootstrap-expect=3

# other nodes with -join=dev-do1
```

When I swich to `gliderlabs/consul:legacy` it works as expected:

```
# curl localhost:8500/v1/catalog/nodes
[{"Node":"dev-do1","Address":"10.246.246.174"},{"Node":"dev-do2","Address":"10.246.246.173"},{"Node":"dev-do3","Address":"10.246.246.177"}]
```

What could be the problem?

**Comment by laktak** [comment]
I was missing `-client` that was not required in legacy.

Could you update your README?

**Comment by dpgeekzero** [comment]
Please, I just got hit by this.

**Comment by yissachar** [comment]
Just to expand on the comment by @laktak, for me this was solved by setting `-client 0.0.0.0`

**Comment by mmkonrad** [comment]
@yissachar is right.... `-client 0.0.0.0` makes consul listen on every available network interface.
Since you haven't started with '--net="host"' (according to your run command) consul won't be reachable unless you use your advertised address
